bridge-token-factory/tests/event_serialization.rs

```diff
@@ -34,18 +35,47 @@ fn real_data_locked_event_serialization() {
 }
 
 fn rand_string(rng: &mut ThreadRng) -> String {
-    (0..rng.gen::<u8>())
-        .map(|_| (0x20u8 + (rng.gen::<f32>() * 96.0) as u8) as char)
-        .collect()
+    const CHARSET: &[u8] = b"abcdefghijklmnopqrstuvwxyz0123456789._-";
+    let string_len = rng.gen_range(2, 64);
+
+    'main: loop {
+        let rand_str: String = (0..string_len)
+            .map(|_| {
+                let idx = rng.gen_range(0, CHARSET.len());
+                CHARSET[idx] as char
+            })
+            .collect();
+
+        let mut last_char_is_separator = true;
+
+        for c in rand_str.chars() {
+            let current_char_is_separator = match c {
+                '-' | '_' | '.' => true,
+                _ => false,
+            };
+
+            if current_char_is_separator && last_char_is_separator {
+                continue 'main;
+            }
+
+            last_char_is_separator = current_char_is_separator;
+        }
+
+        if !last_char_is_separator {
+            return rand_str;
+        }
+    }
```

Comment:
This function is still quite complex to comprehend and makes unnecessary iterations forth and back. I suggest changing it to something like this. Also, please rename the function to `generate_random_near_account_id()` which represents it's utility more clear.

```suggestion
    const CHARSET_WITHOUT_SEPARATORS: &[u8] = b"abcdefghijklmnopqrstuvwxyz0123456789";
    const CHARSET: &[u8] = b"abcdefghijklmnopqrstuvwxyz0123456789._-";
    let string_len = rng.gen_range(2, 64);

    let mut gen_rand_char = |use_separators: bool| -> char {
        let charset = if use_separators {
            CHARSET
        } else {
            CHARSET_WITHOUT_SEPARATORS
        };
        let idx = rng.gen_range(0, charset.len());
        charset[idx] as char
    };

    let is_separator_char = |c: char| -> bool { matches!(c, '-' | '_' | '.') };

    let mut is_last_char_separator = true;
    let mut rand_str: String = (0..string_len)
        .map(|_| {
            let rand_char = gen_rand_char(!is_last_char_separator);
            is_last_char_separator = is_separator_char(rand_char);
            rand_char
        })
        .collect();

    // Make last element non-separator
    if is_last_char_separator {
        rand_str.pop();
        rand_str.push(gen_rand_char(false));
    }

    rand_str
```

Reply:
fixed https://github.com/aurora-is-near/rainbow-token-connector/pull/159/commits/63d8a3f9ee9c4b4b586a123050f2ca391c5164cf